### ISSUE TITLE
Cannot return svg image from lambda endpoint

### ISSUE THREAD

**Issue by aquacash5** [awslabs/aws-lambda-rust-runtime#531]
I am trying to create a lambda service that returns a svg based on a Jira ticket. I have everything I need to create the svg, and I got my lambda service running in AWS. The problem I am having is that the endpoint will only return a base64 encoded version of the data. I am not sure what is going wrong, but I created this example project to show what is going on.

https://github.com/aquacash5/svg-aws-rs-example

## Raw binary response
```rust
async fn function_handler(_event: Request) -> Result<Response<Body>, Error> {
    let svg_doc = create_svg();
    let mut svg_data: Vec<u8> = Vec::new();
    svg::write(&mut svg_data, &svg_doc)?;
    let resp = Response::builder()
        .status(200)
        .header("content-type", "image/svg+xml")
        .body(svg_data.into())
        .map_err(Box::new)?;
    Ok(resp)
}
```

## Convert to string first
```rust
async fn function_handler(_event: Request) -> Result<Response<Body>, Error> {
    let svg_doc = create_svg();
    let mut svg_data: Vec<u8> = Vec::new();
    svg::write(&mut svg_data, &svg_doc)?;
    let svg_str = String::from_utf8(svg_data)?;
    let resp = Response::builder()
        .status(200)
        .header("content-type", "image/svg+xml")
        .body(svg_str.into())
        .map_err(Box::new)?;
    Ok(resp)
}
```

Both versions produce the same result.

**Comment by bnusunny** [comment]
`image/svg+xml` is treated as binary by lambda-http crate. You need to add `BinaryMediaTypes` in your SAM tamplete to configure REST API to work with binary response.  This is in API Gateway's [developer guide](https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-payload-encodings.html#api-gateway-payload-encodings-proxy).

```
  ApiGatewayApi:
    Type: AWS::Serverless::Api
    Properties:
      StageName: prod
      MethodSettings:
        - ResourcePath: /
          HttpMethod: GET
      BinaryMediaTypes:
      - *~1*
```

**Comment by bnusunny** [comment]
Yes, we can. But we could not cover all txt types. Some of them will be treated as binary. Perhaps another configuration to allow users to specify additional text content-types?

**Comment by aquacash5** [comment]
Ah, I thought that returning the `Body:: Binary` and `Body::Text` was supposed to indicate how final response was formatted. From the perspective of a library user, I am telling the you (the library) that I am returning text (or in my case text encoded as binary). But from the link you sent me, it seems that information is either ignored or erased at some point.

Thank you for the help.
